vector/src/main/java/fr/gouv/tchap/activity/TchapRoomCreationActivity.java

```diff
@@ -166,13 +171,35 @@ private void hideKeyboard() {
         imm.toggleSoftInput(InputMethodManager.SHOW_FORCED, 0);
     }
 
-    @OnClick(R.id.rly_hexagon_avatar)
-    void addRoomAvatar() {
+    private void openMediasPicker() {
         Intent intent = new Intent(TchapRoomCreationActivity.this, VectorMediasPickerActivity.class);
         intent.putExtra(VectorMediasPickerActivity.EXTRA_AVATAR_MODE, true);
         startActivityForResult(intent, REQ_CODE_UPDATE_ROOM_AVATAR);
     }
 
+    @OnClick(R.id.rly_hexagon_avatar)
+    void addRoomAvatar() {
+        if (!CommonActivityUtils.checkPermissions(CommonActivityUtils.PERMISSION_CAMERA, this)) {
+            
+        } else {
+            openMediasPicker();
+        }
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+
+        if (requestCode == REQUEST_CODE_PERMISSION_ROOM_DETAILS) {
```

Comment:
You should not use "REQUEST_CODE_PERMISSION_ROOM_DETAILS" but "PERMISSION_CAMERA"